csi/controller.go

```diff
@@ -115,19 +118,122 @@ func (s *OsdCsiServer) ControllerGetCapabilities(
 // ControllerPublishVolume is a CSI API implements the attachment of a volume
 // on to a node.
 func (s *OsdCsiServer) ControllerPublishVolume(
-	context.Context,
-	*csi.ControllerPublishVolumeRequest,
+	ctx context.Context,
+	req *csi.ControllerPublishVolumeRequest,
 ) (*csi.ControllerPublishVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "This request is not supported")
+
+	logrus.Debugf("Calling Controller Publish Volume in the csi controller")
```

Comment:
For every log in the CSI package, please use `clogger.WithContext(ctx).*` instead of `logrus.*`. This include component metadata in the logs as well as correlation ID.